skops/card/_model_card.py

```diff
@@ -704,12 +704,37 @@ def _add_single(self, key: str, val: Formattable | str) -> Section:
         if leaf_node_name in section:
             # entry exists, only overwrite content
             section[leaf_node_name].content = val
+            # if node already existed but was invisible, make it visible
+            section[leaf_node_name].visible = True
         else:
             # entry does not exist, create a new one
             section[leaf_node_name] = Section(title=leaf_node_name, content=val)
 
         return section[leaf_node_name]
 
+    @property
+    def model_diagram(self) -> bool:
+        return self._model_diagram
+
+    @model_diagram.setter
+    def model_diagram(self, value: bool) -> None:
+        self._model_diagram = value
+
+        # If we use the skops template, we know what section to add or remove
+        # when model_diagram changes values. If not, we don't know and thus need
+        # to skip this step.
+        if self.template != Templates.skops.value:
+            return
```

Comment:
As in, instead of returning here, we could raise an error, and if we do, then this PR can be considered to fully fixing the issue IMO.

Reply:
I forgot about this PR, hence the late reply :)

I changed the code to raise an error here, with a hopefully useful error message for the user.

This is still not solving the whole problem though. We have `test_setting_model_diagram_false_other_section`, which is still `xfail`ing. The problem is that even with a skops template, if the user adds a model diagram to a different section, since we don't store that section name, we cannot remove it later on. For that, I see two possible solutions, but I don't like either:

1. Don't allow users to add a model diagram to a different section when they have the skops template (why would they need it twice?) -- but should we really disallow it?
2. We remember where we put stuff, e.g. saving an attribute `_section_name_of_model_diagram`, but that seems to be overly complex for such a niche error.

WDYT?

Reply:
So this makes me realize maybe our abstraction is a bit the issue. We're trying to handle metadata related to the template, but not stored in the template. One way to fix this, would be to allow users to set attributes on the template, such as "collapsed/expanded" for each section. We can still accept a dictionary as a template, but if the template extends the dictionary and includes such information, we apply them to the rendered readme. WDYT?

Reply:
I agree that this smells of incorrect abstraction. But I'm not sure if the proposed solution is the right one. Right now, the template is used to prefill the model card, not anything more. Let's assume we would allow users, as you suggest, to set "collapsed/expanded" on the template. If they use the template that way, and later add a new section, how would they control if that section is collapsed? As it's not in the template, they would have to set it on the card. Now there are two different ways to achieve the same thing.

I think the problem in the discussed case is the conflict between flexibility&customization vs predefined sections. To accommodate the former, we have to allow 0, 1, or multiple model diagrams anywhere in the card. For the latter, we have to assume there either is or isn't a model diagram, and it always sits in the same place. I don't have a good proposal to resolve that conflict.

Regarding this specific problem, it might be better to not have a `model_diagram=True/False` argument on `Card` at all. Instead, just never create it automatically, and if I need it, I have to call `model_card.add_model_plot(section_name)` (the `section_name` argument would be optional with the skops template). If I don't want it anymore, I call  `model_card.delete(section_name)`. IMO that is much cleaner, but it's breaking BC.

Reply:
Sounds doable. For clarification:

> if the user wants to hide the model diagram, they should hide the section

Is your suggestion that if the user sets `model_card.model_diagram = False`, we should try to make it invisible (this PR) or just do nothing (status quo)?